examples/functions-with-included-record-parameters/functions_with_included_record_parameters.bal

```diff
@@ -0,0 +1,83 @@
+import ballerina/io;
+ 
+type Student record {|
+    string firstName;
+    string lastName?;
+|};
+
+type AdmissionDate record {|
+    string year = "2020";
+    string month = "Jan";
+    never admissionNo?;
+|};
+
+type Grades record {|
+    never math?;
+    never physics?;
+    int...;
+|};
+
+// Adding `*` before the parameter name marks the parameter as an included record parameter.
+// The names of the fields in the record type of an included record parameter must be distinct 
+// from each other and also from the names of the other parameters, unless it is an optional field of type never. 
+// If not, it results in a compilation error.
+function printStudentDetails(int admissionNo, *Student student) {
+    string name = student.firstName;
+ 
+    string? lastName = student?.lastName;
+    if lastName is string {
+        name += string ` ${lastName}`;
+    }
+ 
+    io:println("Admission No: ", admissionNo, ", Student Name: ", name);
+}  
+
+function printAdmissionDate(int admissionNo, *AdmissionDate date) {
+    string admissionDate = date.year + "-" + date.month;
+
+    io:println("Admission No: ", admissionNo,
+               ", Admission Date: ", admissionDate);
+}
+
+function printAverage(int math, int physics, *Grades grades) {
+    int totalMarks = math + physics;
+    int count = grades.length() + 2;
+ 
+    foreach int grade in grades {
+        totalMarks += grade;
+    }
+ 
+    io:println("Average Marks: ", totalMarks/count);
+}
+ 
+public function main() {
+
+    Student student = {
+        firstName: "Anna",
+        lastName: "Christina"
+    };
+    
+    // Call the function by passing a positional argument to the `student` included record parameter.
+    printStudentDetails(1001, student);
+
+    // Call the function by passing values for the fields of the included record parameter. 
+    // Need to supply values for the required fields without explicit default values.
+    // If not, accessing an undefined key will result in a compilation error.
```

Comment:
What are we trying to say here. Can we reword this?

It currently reads like 

"If values are not specified for the required fields that do not have default values, accessing an undefined key will result in a compilation error."

Reply:
I tried to tell when accessing key without a value, it gives an error. I removed it.